scripts/zimbra-profile.sh

```diff
@@ -1,3 +1,7 @@
 if [[ -n "$PATH" ]]; then
     export PATH="/opt/zimbra/bin:$PATH"
 fi
+
+if which javac; then
+    export JAVA_HOME=$(dirname $(dirname $(readlink -f $(which javac))))
```

Comment:
Let's change 'which' (an external program) to use the bash built-in 'type' command (I suggest: type -P) in both lines 5 and 6.